src/proxy/pool.rs

```diff
@@ -86,15 +87,13 @@ impl ConnSpawner {
     async fn new_pool_conn(&self, key: WorkloadKey) -> Result<ConnClient, Error> {
         debug!("spawning new pool conn for {}", key);
 
-        let local = self
-            .cfg
-            .enable_original_source
-            .unwrap_or_default()
-            .then_some(key.src);
+        let local = self.original_source.then_some(key.src);
```

Comment:
Ugh yeah I missed this.

Reply:
well back when config was mut I think we changed the meaning of that field from "user intent" to "active state" so it is pretty subtle